var/spack/repos/spack_repo/builtin/packages/strumpack/package.py

```diff
@@ -104,6 +104,8 @@ class Strumpack(CMakePackage, CudaPackage, ROCmPackage):
         depends_on(
             "slate amdgpu_target={0}".format(val), when="+slate amdgpu_target={0}".format(val)
         )
+    for val in CudaPackage.cuda_arch_values:
+        depends_on("slate cuda_arch={0}".format(val), when="+slate cuda_arch={0}".format(val))
```

Comment:
Alternatively you could also leverage the `when` context manager and get rid of the `+cuda+slate` dependency on line 101, e.g.:

```suggestion
    with when("+cuda+slate"):
        for val in CudaPackage.cuda_arch_values:
            depends_on(f"slate +cuda cuda_arch={val}", when=f"cuda_arch={val}")
```